clients/src/main/java/org/apache/kafka/common/requests/TransactionResult.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.common.requests;
+
+public enum TransactionResult {
+    COMMIT((byte) 0), ABORT((byte) 1), UNKNOWN((byte) -1);
+
+    public final byte id;
+
+    TransactionResult(byte id) {
+        this.id = id;
+    }
+
+    public static TransactionResult forId(byte id) {
+        switch (id) {
+            case 0:
+                return COMMIT;
+            case 1:
+                return ABORT;
+            default:
+                return UNKNOWN;
```

Comment:
That was for the actual control record.  This is for the RPC. I think `UNKNOWN` doesn't make sense for the RPC, since by definition we should never write 'UNKNOWN'. I think we should drop it from here.